overlord/fdestate/backend/reseal.go

```diff
@@ -62,6 +62,49 @@ type StateUpdater func(role string, containerRole string, bootModes []string, mo
 
 // ResealKeyForBootChains reseals disk encryption keys with the given bootchains.
 func ResealKeyForBootChains(updateState StateUpdater, method device.SealingMethod, rootdir string, params *boot.ResealKeyForBootChainsParams, expectReseal bool) error {
+	return resealKeys(updateState, method, rootdir,
+		resealInputs{
+			bootChains: params,
+		},
+		resealOptions{
+			ExpectReseal: expectReseal,
+		})
+}
+
+// ResealKeysForSignaturesDBUpdate reseals disk encryption keys for the provided
+// boot chains and an optional signature DB update
+func ResealKeysForSignaturesDBUpdate(
+	updateState StateUpdater, method device.SealingMethod, rootdir string,
+	params *boot.ResealKeyForBootChainsParams, dbUpdate []byte,
+) error {
+	return resealKeys(updateState, method, rootdir,
+		resealInputs{
+			bootChains:        params,
+			signatureDBUpdate: dbUpdate,
+		},
+		resealOptions{
+			ExpectReseal: true,
+			Force:        true,
```

Comment:
Might be a good idea to add a small comment on why we need to force a reseal

Reply:
added